src/lib.rs

```diff
@@ -399,11 +399,20 @@ pub enum SyncParam {
     Bucket { bucket: String, path: String },
 }
 
+impl SyncParam {
+    pub fn new_local(path: String) -> SyncParam {
+        SyncParam::Local { path }
+    }
+    pub fn new_bucket(bucket: String, path: String) -> SyncParam {
+        SyncParam::Bucket { path, bucket }
+    }
+}
+
 impl std::str::FromStr for SyncParam {
     type Err = String;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let s3_format = Regex::new(r"^s3://+?(?P<bucket>[a-zA-Z0-9._-]+)/(?P<prefix>.*)$").unwrap();
+        let s3_format = Regex::new(r"^s3://+?(?P<bucket>[^/]+)/(?P<prefix>.*)$").unwrap();
```

Comment:
playing around with the regex here: https://regex101.com/r/vZL9L9/1 -- I don't think the `+?` after `s3://` is necessary